### ISSUE TITLE
Adaptive Dask Executor workers don't scale up when running many quick tasks

### ISSUE THREAD

**Issue by Adam-D-Lewis** [PrefectHQ/prefect#5483]
## Description

When mapping over a large set of inputs which creates many tasks, an adaptive DaskExecutor will not scale up it's workers when the tasks that are mapping over the inputs are quick.  Scale up does occur when the tasks are slower.

## Reproduction

```python
from dask_gateway import Gateway
from distributed import Client
from prefect import task, Flow
import time
from prefect.executors import DaskExecutor

# Define tasks
@task
def my_quick_task(x):
    return

@task
def my_slow_task(x):
    time.sleep(5)
    return

large_iterable_to_map_over = list(range(100_000))

with Flow('Quick Task Flow') as quick_task_flow:
    my_quick_task.map(large_iterable_to_map_over)

with Flow('Slow Task Flow') as slow_task_flow:
    my_slow_task.map(large_iterable_to_map_over)

# Create new adaptive dask cluster for DaskExecutor
with Gateway('http://localhost:8000') as gateway:
    options = gateway.cluster_options()
    with gateway.new_cluster(options) as cluster, Client(cluster) as client:
        cluster.adapt(1, 5)
        client.wait_for_workers(1)
        my_executor = DaskExecutor(cluster.scheduler_address, client_kwargs={"security": cluster.security})

        # Adaptive Dask Cluster won't scale up
        print('starting quick_task_flow')
        quick_task_flow.run(executor=my_executor)

        # Adaptive Dask Cluster will scale up
        print('starting slow_task_flow')
        slow_task_flow.run(executor=my_executor)
```

## Environment
```json
{
  "config_overrides": {
    "backend": true
  },
  "env_vars": [],
  "system_information": {
    "platform": "Linux-5.10.98-x86_64-with-glibc2.2.5",
    "prefect_backend": "server",
    "prefect_version": "0.15.13",
    "python_version": "3.8.12"
  }
}
```
dask == 2021.08.01
distributed == 2021.08.01
dask-gateway == 0.9.0

**Comment by madkinsz** [comment]
Hi! The scaling here should be fully handled by Dask. Do you see scaling occur if submitting tasks to Dask directly rather than using Prefect?

**Comment by Adam-D-Lewis** [comment]
Yeah, I do see Dask scaling up when using `client.map` directly though it took 40-50 seconds.  I updated the example above, and took note of how long until the additional worker pods show up after the cluster.adapt(1, 5) line, and added what I'm seeing to the comments in the example above.

**Comment by jbbqqf** [comment]
Hi! Triaging older issues this week — I think this one can be closed as out-of-scope for current Prefect.

- The issue is reported against `prefect==0.15.13` (Prefect 1.x) and references `prefect.executors.DaskExecutor` plus the `Flow(...)` class API — both of which have been removed in Prefect 3.x. Today `src/prefect/agent.py` is just a migration stub: ["This surfaces an actionable error message for moved or removed objects in Prefect 3.0 upgrade."](https://github.com/PrefectHQ/prefect/blob/main/src/prefect/agent.py#L1-L3)
- Dask integration has moved out of this repo: [`prefect-archive/prefect-dask`](https://github.com/prefect-archive/prefect-dask) is now under the `prefect-archive` org and is archived. Any remaining Dask-adaptive-scaling work would belong upstream in Dask / `dask-gateway`, not Prefect core.

Would you mind closing this out? If you're still hitting an adaptive-scaling issue on Prefect 3.x with the `DaskTaskRunner`, opening a fresh ticket with a 3.x repro would be the cleanest path.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the file/line and archive-status references above were verified manually against the current `main` branch and the GitHub API.)
